types/cypress-image-snapshot/plugin.d.ts

```diff
@@ -1,5 +1,28 @@
-/// <reference types="Cypress" />
+/* Definitions extracted from Cypress 4.5, which is the minimum
+ * version required by the cypress-image-snapshot.
+ * https://github.com/cypress-io/cypress/blob/v4.5.0/cli/types/index.d.ts#L4513-L4524
+ *
+ * The definitions have been copied in, as DefinitelyTyped does not (yet) support
+ * peerDependencies. This is causing with npms version resolution for consumers.
+ * https://github.com/DefinitelyTyped/DefinitelyTyped/issues/20290#issuecomment-589738371
+ * this has been addressed before on these definitions here
+ * https://github.com/DefinitelyTyped/DefinitelyTyped/pull/41921
+ *
+ * The Cypress interfaces have been stripped down, according to consumed fields
+ * in https://github.com/jaredpalmer/cypress-image-snapshot/blob/master/src/plugin.js
+ */
+type Task = (value: any) => any;
 
-declare function addMatchImageSnapshotPlugin(on: Cypress.PluginEvents, config: Cypress.PluginConfigOptions): void;
+interface Tasks {
+    [key: string]: Task;
+}
+
+interface PluginEvents {
+    (action: 'after:screenshot', fn: (details: {path: string}) => void): void;
+    (action: 'task', tasks: Tasks): void;
+}
+/* Cypress definitions - end */
+
+declare function addMatchImageSnapshotPlugin(on: PluginEvents, config: {}): void;
```

Comment:
Do we want to revert back to `config: any`? I don't think `{}` will allow consumers to pass anything in unless that was the intention, it's been a while since I've looked at the API

Reply:
as far as i read, it looks like no fields from config are consumed by the plugin (https://github.com/jaredpalmer/cypress-image-snapshot/blob/cf42ab464de9f391507d50e4c6056a206b119b24/src/plugin.js#L137-L138 see function definitions). it is also not the config for the plugin, but `Cypress.PluginConfigOptions`, an argument from the cypress setup function.

as far as i understand `{}` should be equivalent to `object`. as both are a subset of `Cypress.PluginConfigOptions` it should allow to pass in any object, but not a `number` or `string`.